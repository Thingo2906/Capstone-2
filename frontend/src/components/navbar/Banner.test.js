import React from "react";
import Banner from "./Banner";
import MovieApi from "../../api/api";

import UserContext from "../../auth/UserContext";
import { render } from "@testing-library/react";

// Mock functions for required props
const mockAddMovieToList = jest.fn();
const mockRemoveMovieFromList = jest.fn();
const mockFetchData = jest.fn();


// Mock UserContext with addedMovies property
const mockUserContext = {
  addedMovies: [], // we can customize this value as needed for the test
};

describe("Banner Component", function(){
    test("renders without crashing", () => {
        mockFetchData.mockResolvedValue({
            data: {
              results: [
                {
                  id: 1,
                  name: "Movie Name",
                  backdrop_path: "sample-image.jpg",
                  overview: "Movie overview...",
                },
              ],
            },
          });
  

         render(
            <UserContext.Provider value={mockUserContext}>
              <Banner
                fetchData={mockFetchData}
                addMovieToList={mockAddMovieToList}
                removeMovieFromList={mockRemoveMovieFromList}
              />
            </UserContext.Provider>
        );
    });
    test("matches snapshot", function () {
      mockFetchData.mockResolvedValue({
         data: {
           results: [
             {
               id: 1,
               name: "Movie Name",
               backdrop_path: "sample-image.jpg",
               overview: "Movie overview...",
             },
           ],
         },
       });
      const { asFragment } = render(
        <UserContext.Provider value={mockUserContext}>
          <Banner
            fetchData={mockFetchData}
            addMovieToList={mockAddMovieToList}
            removeMovieFromList={mockRemoveMovieFromList}
          />
        </UserContext.Provider>
      );
      expect(asFragment()).toMatchSnapshot();
    });
    
  });
