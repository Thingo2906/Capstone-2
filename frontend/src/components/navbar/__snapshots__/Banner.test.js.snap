// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner Component matches snapshot 1`] = `
<DocumentFragment>
  <header
    class="banner"
    style="background-size: cover; background-image: url(http://image.tmdb.org/t/p/original/); background-position: center center;"
  >
    <div
      class="banner-contents"
    >
      <h1
        class="banner-title"
      />
      <div
        class="banner-buttons"
      >
        <button
          class="banner-button"
        >
          Play
        </button>
        <button
          class="banner-button"
        >
          + My List
        </button>
      </div>
      <h1
        class="banner-description"
      />
    </div>
    <div
      class="banner-fadebottom"
    />
  </header>
</DocumentFragment>
`;
